packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -46,8 +57,13 @@ export type InsightsProps<
   $$internal?: boolean;
 };
 
-const ALGOLIA_INSIGHTS_SRC =
-  'https://cdn.jsdelivr.net/npm/search-insights@2.3.0/dist/search-insights.min.js';
+const ALGOLIA_INSIGHTS_VERSION = '2.4.0';
+const ALGOLIA_INSIGHTS_SRC = `https://cdn.jsdelivr.net/npm/search-insights@${ALGOLIA_INSIGHTS_VERSION}/dist/search-insights.min.js`;
+
+export type InsightsClientWithGlobals = InsightsClient & {
+  shouldAddScript?: boolean;
+  version?: string;
```

Comment:
`version` is available in `InsightsClient` if you update `search-insights` to v2.4.0.

Reply:
Makes sense, I'll leave it as optional for now and update the dependency in a separate PR